DEVELOPER_GUIDE.md

```diff
@@ -422,6 +423,79 @@ Output plugins are defined in [flb_output.h](https://github.com/fluent/fluent-bi
 
 The [stdout plugin](plugins/out_stdout) is very simple; review its code to understand how output plugins work.
 
+#### Config Maps
+
+Config maps are an improvement to the previous Fluent Bit exposed API that was used by plugins to read configuration values. The new config maps feature warns the user in case of getting an unknown configuration key and reduces risk of  bad configuration due to typos or deprecated properties names. They are also important  so that in the future dynamic reload support can be implemented.
+
+There are various types of supported configuration types. Full list available [here](https://github.com/fluent/fluent-bit/blob/master/include/fluent-bit/flb_config_map.h#L29). The most used ones are :
+
+| Type                   | Description           | 
+| -----------------------|:---------------------:| 
+| FLB_CONFIG_MAP_INT     | Represents integer data type | 
+| FLB_CONFIG_MAP_BOOL    | Represents boolean data type | 
+| FLB_CONFIG_MAP_DOUBLE  | Represents a double |
+| FLB_CONFIG_MAP_SIZE    | Converts string size to integer |
```

Comment:
nit: When would I use SIZE vs INT?   That's not clear from this description; is size converted into a `size_t`? (I am personally new to config maps).

Reply:
In my understaning, it provides string::size_type as an integer datatype large enough to represent any possible string size.